### PR TITLE
`azurerm_data_factory_integration_runtime_azure` - remove `Computed` from `cleanup_enabled`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description
The `cleanup_enabled` is introduced by this [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/13222/files#diff-a6b52427901543cc50c2f2f87d94ea81ef47ea1581c9507b4b72468888adc9b5). This [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/13222/files#diff-a6b52427901543cc50c2f2f87d94ea81ef47ea1581c9507b4b72468888adc9b5) sets the default value "true" in the create operation and adds Computed to omit the difference. I think it's incorrect and we should set the default value in the property schema.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_data_factory_integration_runtime_azure` - remove `Computed` from `cleanup_enabled`

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/datafactory/data_factory_integration_runtime_azure_resource.go [hunk 1]
```diff
@@ -272,11 +283,18 @@ func expandDataFactoryIntegrationRuntimeAzureComputeProperties(d *pluginsdk.Reso
...
+	var cleanup bool
+	if features.FourPointOhBeta() {
+		cleanup = d.Get("cleanup_enabled").(bool)
+	} else {
+		cleanup = true
+		// nolint staticcheck
+		if v, ok := d.GetOkExists("cleanup_enabled"); ok {
+			cleanup = v.(bool)
+		}
 	}
```
Comment (stephybun) [inline internal/services/datafactory/data_factory_integration_runtime_azure_resource.go:296]: We were defaulting this to `true` in the expand before so we should still maintain that behaviour for now

```suggestion
	cleanup := true
	if features.FourPointOhBeta() {
		cleanup = d.Get("cleanup_enabled").(bool)
	} else {
		// nolint staticcheck
		if v, ok := d.GetOkExists("cleanup_enabled"); ok {
			cleanup = v.(bool)
		}
	}
```

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/datafactory/data_factory_integration_runtime_azure_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/datafactory/data_factory_integration_runtime_azure_resource.go]: Updated

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comments. I updated PR. Please take another look. Below is the test result I just now triggered.

image

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM 👍

**Pull request merged**